### DISCUSSION TITLE
Question about UUID

### DISCUSSION THREAD

**Discussion by shoffmanaimsplus** [brianvoe/gofakeit#204] [Q&A]
Greetings, thank you for writing this library. It's definitely saving me a lot of time. With that being said, I do have a question. It's quite possible that I'm using this incorrectly, but I was trying to write my own custom function that mirrors UUID but instead of returning a string, instead returns a custom data type (in this case, a Google uuid.UUID).  Given the following:

`
/* Yes I realize that I'm not using r, but I have a global that I'm using for ourfaker which is initialized
to the correct seed */
```
func getUuidBinary(r *rand.Rand) uuid.UUID {
	str := ourfaker.UUID()
	var guid uuid.UUID
	guid, _ = uuid.Parse(str)
	return guid
}

func addGuidLookup() {
	gofakeit.AddFuncLookup("guid", gofakeit.Info{
		Display:     "Guid",
		Category:    "Uuid Derivative",
		Description: "Guid",
		Example:     "590c1440-9888-45b0-bd51-a817ee07c3f2",
		Output:      "uuid.UIID",
		Generate: func(r *rand.Rand, m *gofakeit.MapParams, info *gofakeit.Info) (interface{}, error) {
			return getUuidBinary(r), nil
		},
	})
}

type MyStruct struct {
    My_guid          uuid.UUID `fake:"{guid}"
}

s := &MyStruct{}

gofakeit.Struct(s)
```
`

When all is said and done, s.My_Guid is all zeroes, though if I put a breakpoint in getUuidBinary I'm returning the correct value. If there's
a better way to do this, I'm all ears (and yes, I'm aware of New() in Google's package, but I want to use the gofakeit version because I can get repeatability by using the same seed.

Thanks in advance!

**Comment by brianvoe** [comment]
I recently did an update to potentially fix that from a pr that someone submitted. Can you confirm you are using the more recent version? If you are, can you please try to give me a full example that i can run to test where its not binding to your type?

**Comment by zergeborg** [comment]
@brianvoe I am experiencing a similar issue in v7@v7.0.2. For example:
```
package main

import (
	"encoding/json"
	"fmt"

	"github.com/brianvoe/gofakeit/v7"
	"github.com/gofrs/uuid"
)

type Info struct {
	SequenceNumber       uuid.UUID `json:"SequenceNumber,omitempty" fake:"{uuid}"`
}

func main() {
	var b Info
	gofakeit.Struct(&b)
	js, _ := json.Marshal(b)

	fmt.Println(string(js))

	fmt.Println(gofakeit.UUID())
}
```

This prints out the following:
```
{"SequenceNumber":"00000000-0000-0000-0000-000000000000"}
c2faff54-ec69-4042-a10c-820411504127
```

**Reply by zergeborg** [reply]
Playground link:
https://go.dev/play/p/ZnVNhFobm6z

**Reply by zergeborg** [reply]
I also realized that if I change the `uuid.UUID` type to string, then the JSON string is correct

**Reply by zergeborg** [reply]
OK, so after even more digging and debugging I can see that `GetFuncLookup` in `struct.go` returns this:
```
v7.Info {Display: "UUID", Category: "misc", Description: "128-bit identifier used to uniquely identify objects or entities in computer systems", Example: "590c1440-9888-45b0-bd51-a817ee07c3f2", Output: "string", ContentType: "text/plain", Params: []github.com/brianvoe/gofakeit/v7.Param len: 0, cap: 0, nil, Any: interface {} nil, Generate: github.com/brianvoe/gofakeit/v7.addMiscLookup.func1}
```

Is this correct? I cannot tell, because I am new to this library. In any case, when the code moves on to `info.Generate(f, mapParams, info)` it generates a correct value (`interface {}(string) "331e842b-ae0e-4461-af3b-24d10d7059f1"`)

But later when the code "Check(s) if field kind is the same as the expected type" it returns with:
`errors.New("field kind " + fieldElem.Kind().String() + " does not match expected kind " + v.Kind().String())`

Is this behavior expected? Again, I cannot tell.

**Comment by brianvoe** [comment]
Gofakeit uuid returns a string. Is your uuid.UUID w string? If not that's why
